app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+    def die(self) -> None:
+        if self in Animal.alive and self.health <= 0:
```

Comment:
The condition `self.health <= 0` should be checked whenever the health changes, not just in the `die` method. Consider calling `die()` after any action that modifies health, such as in the `bite` method.